browser/ui/webui/brave_sanitized_image_source.h

```diff
@@ -0,0 +1,82 @@
+// Copyright (c) 2025 The Brave Authors. All rights reserved.
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this file,
+// You can obtain one at https://mozilla.org/MPL/2.0/.
+
+#ifndef BRAVE_BROWSER_UI_WEBUI_BRAVE_SANITIZED_IMAGE_SOURCE_H_
+#define BRAVE_BROWSER_UI_WEBUI_BRAVE_SANITIZED_IMAGE_SOURCE_H_
+
+#include <memory>
+
+#include "base/memory/scoped_refptr.h"
+#include "base/memory/weak_ptr.h"
+#include "base/sequence_checker.h"
+#include "content/public/browser/url_data_source.h"
+#include "services/data_decoder/public/cpp/data_decoder.h"
+#include "services/data_decoder/public/cpp/decode_image.h"
+#include "url/gurl.h"
+
+class Profile;
+class SkBitmap;
+
+namespace network {
+class SharedURLLoaderFactory;
+class SimpleURLLoader;
+}  // namespace network
+
+// Brave versions of SanitizedImageSource available via
+// chrome://brave-image/?url=<image_url>&target_size=<width>x<height>
```

Comment:
I think we should note that only one dimension is guaranteed to fit in `target_size`